scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamFunctions.scala

```diff
@@ -22,4 +25,31 @@ object BeamFunctions {
     def processElement(c: DoFn[A, B]#ProcessContext): Unit =
       c.output(f(c.element()))
   }
+
+  case class MapSideAggregator[K, V](size: Int, semigroup: Semigroup[V]) extends DoFn[KV[K, V], KV[K, V]] {
+    var cache: SummingCache[K, V] = _
+    @StartBundle
+    def startBundle(): Unit = {
+      cache = new SummingCache[K, V](size)(semigroup)
+    }
+    @ProcessElement
+    def processElement(c: DoFn[KV[K, V], KV[K, V]]#ProcessContext): Unit = {
+      val key = c.element().getKey
+      val value = c.element().getValue
+      val evicted = cache.put(Map((key, value)))
+      evicted match {
+        case Some(m) => m.foreach(x => c.output(KV.of(x._1, x._2)))
+        case None => ()
+      }
+    }
+    @FinishBundle
+    def finishBundle(c: DoFn[KV[K, V], KV[K, V]]#FinishBundleContext): Unit = {
+      val evicted = cache.flush
+      evicted match {
+        case Some(m) =>
+          m.foreach(x => c.output(KV.of(x._1, x._2), BoundedWindow.TIMESTAMP_MIN_VALUE, GlobalWindow.INSTANCE))
```

Comment:
Outside processElement Beam has no notion of window or timestamp to output elements, but also Beam does not set windows if a windowFn is not specified, so everything should remain in GlobalWindow, also Beam set all elements to TimestampMinValue if a timestamp is not provided manually to each element, so we are also setting everting to min value here